Bugfix/diacritic xref

glasseyes opened the pull request:
The cross reference ids weren't being normalised when writen out by LiftWriter.
This created a problem in WeSay where SFM export can't find the entry being referenced and puts the full id reference in instead of the entry. There will be a migration in WeSay in 1.6 to fix existing broken entries

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/sillsdev/libpalaso/587)
<!-- Reviewable:end -->

megahirt commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> see comment below

---

Reviewed 1 of 2 files at r1.
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Palaso.DictionaryServices.Tests/Lift/LiftWriterTests.cs, line 366 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/587#-KxlrEY-8tfEGMuTfZF4:-KxlrEY-8tfEGMuTfZF5:b-ei7fez) ([raw file](https://github.com/sillsdev/libpalaso/blob/b6e06220054403b8afcda1fa4a674aa2c51f555e/Palaso.DictionaryServices.Tests/Lift/LiftWriterTests.cs#L366)):*
> ```Smalltalk
> 
> 		[Test]
> 		public void EntryWithRelationToDiacriticWord()
> ```

In general test names should be written

`MethodOrObjectUnderTest]_Environment_ExpectedResult`

So perhaps rename to 

`public void LiftWriter_EntryWithRelationToDiacriticWord_LiftReferenceIsNFC`

or something that sounds good to you

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/587#-:-Kxlrx6a5clDDdoj2LEW:b-edxffw)*
<!-- Sent from Reviewable.io -->

glasseyes commented:
Review status: 1 of 2 files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Palaso.DictionaryServices.Tests/Lift/LiftWriterTests.cs, line 366 at r1](https://reviewable.io:443/reviews/sillsdev/libpalaso/587#-KxlrEY-8tfEGMuTfZF4:-KxpYcYgga8mpU9OgZUG:b-akr3t3) ([raw file](https://github.com/sillsdev/libpalaso/blob/b6e06220054403b8afcda1fa4a674aa2c51f555e/Palaso.DictionaryServices.Tests/Lift/LiftWriterTests.cs#L366)):*
<details><summary><i>Previously, megahirt (Chris Hirt) wrote…</i></summary><blockquote>

In general test names should be written

`MethodOrObjectUnderTest]_Environment_ExpectedResult`

So perhaps rename to 

`public void LiftWriter_EntryWithRelationToDiacriticWord_LiftReferenceIsNFC`

or something that sounds good to you
</blockquote></details>

No other tests in the file conform to that and likewise for the majority if not all tests in libpalaso. But, while I was there, I changed all the tests in the file to have a LiftWriter_ prefix to make them easier to spot, and added some comments to say what the expected result is.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/sillsdev/libpalaso/587)*
<!-- Sent from Reviewable.io -->
